driver-mapping/src/main/java/com/datastax/driver/mapping/PropertyMapper.java

```diff
@@ -15,182 +15,28 @@
-    }
-
-    @SuppressWarnings("unchecked")
-    <A extends Annotation> A annotation(Class<A> annotationClass) {
-        return (A) annotations.get(annotationClass);
-    }
-
-    boolean isComputed() {
-        return hasAnnotation(Computed.class);
-    }
-
-    boolean isTransient() {
-        return hasAnnotation(Transient.class);
-    }
-
-    boolean isPartitionKey() {
-        return hasAnnotation(PartitionKey.class);
-    }
-
-    boolean isClusteringColumn() {
-        return hasAnnotation(ClusteringColumn.class);
-    }
-
-    private String inferColumnName() {
-        if (isComputed()) {
-            return annotation(Computed.class).value();
-        }
-        boolean caseSensitive = false;
-        String columnName = propertyName;
-        if (hasAnnotation(Column.class)) {
-            Column column = annotation(Column.class);
-            caseSensitive = column.caseSensitive();
-            if (!column.name().isEmpty())
-                columnName = column.name();
-        } else if (hasAnnotation(com.datastax.driver.mapping.annotations.Field.class)) {
-            com.datastax.driver.mapping.annotations.Field udtField = annotation(com.datastax.driver.mapping.annotations.Field.class);
-            caseSensitive = udtField.caseSensitive();
-            if (!udtField.name().isEmpty())
-                columnName = udtField.name();
-        }
-        return caseSensitive ? Metadata.quote(columnName) : columnName.toLowerCase();
-    }
-
-    @SuppressWarnings("unchecked")
-    private TypeToken<Object> inferJavaType() {
-        Type type;
-        if (getter != null)
-            type = getter.getGenericReturnType();
-        else
-            type = field.getGenericType();
-        return (TypeToken<Object>) TypeToken.of(type);
-    }
-
-    private int inferPosition() {
-        if (isPartitionKey()) {
-            return annotation(PartitionKey.class).value();
-        }
-        if (isClusteringColumn()) {
-            return annotation(ClusteringColumn.class).value();
-        }
-        return -1;
-    }
-
-    private TypeCodec<Object> createCustomCodec() {
-        Class<? extends TypeCodec<?>> codecClass = getCustomCodecClass();
-        if (codecClass.equals(Defaults.NoCodec.class))
-            return null;
-        @SuppressWarnings("unchecked")
-        TypeCodec<Object> instance = (TypeCodec<Object>) ReflectionUtils.newInstance(codecClass);
-        return instance;
-    }
-
-    private Class<? extends TypeCodec<?>> getCustomCodecClass() {
-        Column column = annotation(Column.class);
-        if (column != null)
-            return column.codec();
-        com.datastax.driver.mapping.annotations.Field udtField = annotation(com.datastax.driver.mapping.annotations.Field.class);
-        if (udtField != null)
-            return udtField.codec();
-        return Defaults.NoCodec.class;
-    }
-
-    @Override
-    public String toString() {
-        return propertyName;
-    }
+public interface PropertyMapper {
```

Comment:
My intention was to distinguish two populations: "regular" "advanced" users. Regular users would typically use `DefaultPropertyMapper` and just configure the strategies associated with it. Advanced ones would rather extend it or re-implement it completely.
The test that completely bypasses reflection has indeed been removed, not because it doesn't work anymore, but because it's imo something we should leave for the advanced users to to. But please trust me on this, I have tried, and it works (it's just a considerable amount of code to write).